src/entry.rs

```diff
@@ -525,17 +529,28 @@ impl<'a> EntryFields<'a> {
 
         // Ensure we write a new file rather than overwriting in-place which
         // is attackable; if an existing file is found unlink it.
-        fn open(dst: &Path) -> io::Result<std::fs::File> {
-            OpenOptions::new().write(true).create_new(true).open(dst)
+        let mode = self.header.mode();
+        fn open(dst: &Path, _mode: &io::Result<u32>) -> io::Result<std::fs::File> {
```

Comment:
This should not be named `_mode`, but rather it should be called `mode`. I understand it's not used on all platforms, that warning should be fixed via method that does not involve changing the variable name.

Reply:
allow(unused_variables) done instead. I had used _mode as it was narrower ,whereas allowing unused variables will permit other unused variables in future, at least as I understand it.